File: src/modules/role/domain/dto/role_create_dto.py
from ninja import Schema
from typing import List
from src.modules.role.domain.enums.role_references_enum import RoleReferences
from src.shared.domain.enums.status_enum import Status


class RoleCreateDTO(Schema):
    name: str
    is_default: bool
    reference: RoleReferences
    organization_id: int | None = None
    company_id: int | None = None
    status: Status
    abilities_ids: List[int] | None = None

File: src/modules/role/domain/dto/role-create.dto.ts
import { EStatus } from '@/shared/domain/enums/status.enum';
import { ERoleReference } from '../enums/role-reference.enum';

export interface RoleCreateDto {
  name: string;
  reference: ERoleReference;
  organizationId?: number;
  companyId?: number;
  status: EStatus;
  abilitiesIds: number[];
}
